### ISSUE TITLE
Script editor indentation should be 2 spaces

### ISSUE THREAD

**Issue by theonlygusti** [Tampermonkey/tampermonkey#402]
Script editor should use default 2 space indentation, as this is now standard js, but with an option to change.

**Comment by derjanb** [comment]
> should use default 2 space indentation, as this is now standard js

Who defines this standard? w3schools, ESLint as well as this popular style guide are using 4 spaces:

https://www.w3schools.com/js/js_conventions.asp
http://eslint.org/docs/rules/indent.html
http://javascript.crockford.com/code.html

> but with an option to change

Set the config mode to "Advanced" and change "Indentation Width" at the editor section to 2.

**Comment by tophf** [comment]
[StandardJS](https://standardjs.com/) is the actual name of a coding style :-)

**Comment by theonlygusti** [comment]
Google (for whom your extension is written), Mozilla, and GitHub all use _and_ recommend 2-space indentation for JavaScript.

Thanks for showing me how I can change this though.

**Comment by theonlygusti** [comment]
Also did you seriously use w3schools as a authority? Yuck.

**Comment by derjanb** [comment]
> Also did you seriously use w3schools as a authority? Yuck.

It's one of the first matches when searching Google for "javascript indentation". And Google know what's important. 😁

> Google (for whom your extension is written), [...] use and recommend 2-space indentation for JavaScript.

But the Chrome developer tools are using 4 space indentation when auto-formatting code. 🤓

> StandardJS is the actual name of a coding style :-)

The code styling topic is difficult and almost everybody has a different opinion. 
The StandardJS coding style looks good, except the missing semicolons, the 2 space indentation and the very ugly space character after the "function" keyword. 😲 Credits for not explicitly limiting the line length. There is nothing worse than to have a small band of code at the first third of your editor window. But that's another story.

Long story short, for my monitor resolution, DPI and font size a 4 spaces are perfect to outline the indentation without wasting too much space. It's also not really unusual and since it's configurable I'll not change to default in the hope that it might convince more people to use it. 😉

**Comment by tophf** [comment]
On a side note, I also dislike the missing semicolons and spaces after function keyword. Previously I also disliked the 2-space indent, but now I like it. It also allows to open more auxiliary panels in the editor I use primarily (not applicable to TM though).
